rhine-terminal/TerminalSimple.hs

```diff
@@ -0,0 +1,100 @@
+{- | Example application for the @rhine-terminal@ library. -}
+
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+import Prelude hiding (putChar)
+
+-- rhine
+import Control.Monad.Schedule ()
+
+import Data.Text (Text)
+import qualified Data.Text as T
+import System.Terminal
+import System.Terminal.Internal
+import FRP.Rhine
+import System.IO hiding (putChar)
+
+import FRP.Rhine.Terminal (TerminalEventClock(..))
+import System.Exit (exitSuccess)
+
+-- Clocks
+
+data Input = Char Char
+           | Space | Backspace | Enter
+           | Exit
+
+type InputClock = SelectClock (TerminalEventClock LocalTerminal) Input
+
+inputClock :: LocalTerminal -> InputClock
+inputClock term = SelectClock { mainClock = TerminalEventClock term, select = selectKey }
+  where
+    selectKey :: Tag (TerminalEventClock LocalTerminal) -> Maybe Input
+    selectKey = \case
```

Comment:
Rename to `select`

Reply:
Sounds good. Also added `NamedFieldPuns` to make it more terse.